src/util.ts

```diff
@@ -1,5 +1,5 @@
 export const isClient = typeof window === 'object';
 
-export const on = (obj: any, ...args: any[]) => obj.addEventListener(...args);
+export const on = (obj: any, ...args: any[]) => obj && obj.addEventListener(...args);
```

Comment:
Not sure if these functions should be touched at all. With your changes in some cases handler bind will silently fail and evelopment can be hardened. Better to have error in that case.

You should leave only changes in `src/useMediaDevices.ts`, update the PR and ill merge it.

Reply:
As you forbid to make changes in your fork repo it should be like so:

**`useMediaDevices`:**
```typescript
import { useEffect, useState } from 'react';
import { noop, off, on } from './util';

const useMediaDevices = () => {
  const [state, setState] = useState({});

  useEffect(() => {
    if (!navigator.mediaDevices) return;

    let mounted = true;

    const onChange = () => {
      navigator.mediaDevices
        .enumerateDevices()
        .then(devices => {
          mounted &&
            setState({
              devices: devices.map(({ deviceId, groupId, kind, label }) => ({ deviceId, groupId, kind, label })),
            });
        })
        .catch(noop);
    };

    on(navigator.mediaDevices, 'devicechange', onChange);
    onChange();

    return () => {
      mounted = false;
      off(navigator.mediaDevices, 'devicechange', onChange);
    };
  }, []);

  return state;
};

export default useMediaDevices;
```

**`util`**
```typescript
export const isClient = typeof window === 'object';

export const on = (obj: any, ...args: any[]) => obj.addEventListener(...args);

export const off = (obj: any, ...args: any[]) => obj.removeEventListener(...args);

export const noop = () => {};
```